### PR TITLE
Add ML Model Development

### PR DESCRIPTION
**Pull request by TempixTL**
This adds the script that I've been using to develop the ML model. I'd really appreciate a quick pair of eyes to check for best practices and small mistakes.

### CODE REVIEW

File: ml/main.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    
+    while len(encoded) < max_corpus_length:
+      encoded.append(0)
+
+    return encoded
+
+  calorie_data['input'] = calorie_data.apply(lambda row: input_for_row(row), axis=1)
+  return calorie_data
+
+
```
Comment (timmy-ch) [inline ml/main.py:86]: Can you split out the library and the main function into two different files? The main should be executed as

`if "__name__" == "__main__":`

while the library should also have some test functions to make sure they work as intended.

File: ml/main.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+corpus = calorie_data['description']
+vocab_size = num_unique_words(corpus)
+tokenized_corpus = corpus.map(lambda desc: desc.split(' '))
+max_corpus_length = max_sequence_length(tokenized_corpus)
+calorie_data = add_input_labels(calorie_data, vocab_size, max_corpus_length)
+
+# %% create model
+model = keras.Sequential([
+  keras.layers.Embedding(vocab_size, int(vocab_size**(1/4)), input_length=max_corpus_length),
+  keras.layers.Flatten(),
```
Comment (timmy-ch) [inline ml/main.py:103]: Probably want more intermediate layers here, yeah?

File: ml/main.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+model.summary()
+model._layers = [layer for layer in model._layers if not isinstance(layer, dict)] # workaround for bug in keras.util.plot_model
+keras.utils.plot_model(model, show_layer_names=False, show_shapes=True)
+
+# %% split dataset
+train_frac = 0.6
+validation_frac = 0.2
+test_frac = 0.2
+
+train_set, validation_set, test_set = np.split(
```
Comment (timmy-ch) [inline ml/main.py:123]: probably better to use scikit_learn, specifically https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html. To create these three sets, you can just run this function twice

File: ml/main.py (hunk 1 above)
Comment (TempixTL) [reply ml/main.py]: Splitting library and main functions is addressed in https://github.com/googleinterns/amaranth/pull/2/commits/c69201b67a858f0857cad436d771eaca0636aaca

File: ml/main.py (hunk 3 above)
Comment (TempixTL) [reply ml/main.py]: Fixed in https://github.com/googleinterns/amaranth/pull/2/commits/5cc2a688f69aba3abbe2f2089a7fd640eaeb82e8

File: ml/main.py (hunk 1 above)
Comment (TempixTL) [reply ml/main.py:71]: Adding tests for library functions was addressed in 737adebe010615df97234b03d89625f856d959c8

File: ml/main.py (hunk 2 above)
Comment (TempixTL) [reply ml/main.py]: Added a couple Dense layers with sigmoid activation in https://github.com/googleinterns/amaranth/pull/2/commits/db41980dda44b6d0bc3acd54f3ce27d7935a9e54:

https://github.com/googleinterns/amaranth/blob/db41980dda44b6d0bc3acd54f3ce27d7935a9e54/ml/main.py#L72-L79

However, the models performance didn't change by more than a percentage point in precision and recall. Is there some other layer type/activation function/layer config that I should try? I was just playing around with different values and things, not sure if there's any kind of rule of thumb
